pkgs/development/compilers/alan2/default.nix

```diff
@@ -0,0 +1,38 @@
+{ stdenv, lib, fetchFromGitHub
+}:
+
+let
+  pname = "alan2";
+  version = "2.8.7";
+
+in stdenv.mkDerivation {
+
+  name = "${pname}-${version}";
+  inherit pname version;
+
+  src = fetchFromGitHub {
+    owner = "alan-if";
+    repo = "alan";
+    rev = "62c01513a74a52b9d3684dbfdb956202db2ae37e";
```

Comment:
That commit is not part of the repository. It is not allowed to fetch such commits. https://github.com/alan-if/alan/commit/62c01513a74a52b9d3684dbfdb956202db2ae37e

Reply:
It's currently in a pull request and can be fetched. Are you saying it's against policy, or that there's some extra sandboxing that I'm not using when I build it on my machine?

Reply:
My PR has been merged now so I've updated the ref to one that's actually on a branch.

Reply:
> It's currently in a pull request and can be fetched

GitHub allows fetching every commit of every fork. Just change the owner to your fork and everything is fine.

Reply:
Useful to know. Thanks.

In this case, though, the PR has now been merged so there's no need to mention a fork. I changed the ref to point to the merge commit on the `v2_x_branch` in the original repo.